app/src/main/java/com/khuttun/notificationnotes/NotificationMgr.java

```diff
@@ -65,16 +90,7 @@ public void setNotification(NotificationNote note)
 
     public void setGroupNotification(ArrayList<NotificationNote> notes)
     {
-        ArrayList<NotificationNote> visibleNotes = new ArrayList<>();
-        for (int i = 0; i < notes.size(); ++i)
-        {
-            NotificationNote n = notes.get(i);
-            if (n.isVisible)
-            {
-                visibleNotes.add(n);
-            }
-        }
-
+        ArrayList<NotificationNote> visibleNotes = getVisibleNotes(notes);
         if (Globals.LOG) Log.d(Globals.TAG, "Group notification: visible notes " + visibleNotes.size());
 
         switch (visibleNotes.size())
```

Comment:
The "case 0" and "case 1" don't really work when using the grouped notifications. Would it make sense to leave this switch-case to be used only when groupAPI == false, and always call displayGroupNotifications when it's true?

Also, now you are using GROUP_NOTIF_ID in cases 0 and 1, and SUMMARY_NOTIF_ID in default case. That will not work. Consider should you just remove GROUP_NOTIF_ID?